frontend/src/lib/components/PropertyFilters/components/OperatorValueSelect.tsx

```diff
@@ -37,12 +39,16 @@ export function OperatorValueSelect({
     allowQueryingEventsByDateTime,
 }: OperatorValueSelectProps): JSX.Element {
     const [currentOperator, setCurrentOperator] = useState(operator)
+    const { propertyDefinitions } = useValues(propertyDefinitionsModel)
 
-    const operators = allowQueryingEventsByDateTime
-        ? (Object.keys(operatorMap) as Array<PropertyOperator>)
-        : (Object.keys(operatorMap).filter(
-              (o) => o !== 'is_date_before' && o !== 'is_date_after'
-          ) as Array<PropertyOperator>)
+    const propertyDefinition = propertyDefinitions.find((pd) => pd.name === propkey)
+
+    const operators =
+        allowQueryingEventsByDateTime && propertyDefinition?.property_type == 'DateTime'
+            ? (Object.keys(operatorMap).filter((o) => o !== 'gt' && o !== 'lt') as Array<PropertyOperator>)
+            : (Object.keys(operatorMap).filter(
+                  (o) => o !== 'is_date_before' && o !== 'is_date_after'
+              ) as Array<PropertyOperator>)
```

Comment:
> let's make operatorMap a Record<PropertyOperator, string> instead of Record<string, string>

That ballooned into quite a lot of changes that felt off-topic for this PR 

So I've split into two different named operator maps instead of filtering one. It occurs to me that once we know more types of more properties we can be more clever with this